src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt

```diff
@@ -91,14 +90,21 @@ class AutoFollowTask(id: Long, type: String, action: String, description: String
     }
 
     private fun addRetryScheduler() {
+        log.debug("Adding retry scheduler")
         if(retryScheduler != null && !retryScheduler!!.isCancelled) {
             return
         }
-        retryScheduler = try {
-            threadPool.schedule({ failedIndices.clear() }, replicationSettings.autofollowRetryPollDuration, ThreadPool.Names.GENERIC)
+         try {
+            retryScheduler = threadPool.schedule(
+                    {
+                        log.debug("Clearing failed indices to schedule for the next retry")
+                        stat.failedIndices.clear()
```

Comment:
If we're clearing failed indices here, won't we still keep retrying same failing indices ?

Reply:
Idea is to re-try the failed indices as these can fail due to various reasons (security roles update etc) and once the issue is resolved, autofollow will be able to retry in the next run(1hr of retry duration) after clearing these indices.

Reply:
> autofollow will be able to retry in the next run(1hr of retry duration)
is not 1 hr a lot of time to retry after ?

Reply:
Above setting is configurable and can be relaxed if required.